### PR TITLE
Restructure project and document Conan-based build

### PR DESCRIPTION
**Pull request by thejohnfreeman**
I removed all the CI-related files. They seem broken and unused. I can add GitHub workflows if we really want.

### CODE REVIEW

File: README.md
Comment (ximinez) [inline README.md:25]: Maybe add `./validator-keys --unittest` to encourage people to run the tests to ensure the build worked correctly.

**Review by ximinez — APPROVED** [review]
It works!

**Review by ximinez — CHANGES_REQUESTED** [review]
I got excited and spoke too soon. It doesn't build in Windows.
```
$ cmake -DCMAKE_TOOLCHAIN_FILE=../../conan.msvc19/conan_toolchain.cmake -G"Visual Studio 16 2019" ../../..
-- Using Conan toolchain: C:/Dev/ripple-offline-tool/build/conan.msvc19/conan_toolchain.cmake
CMake Error at build/conan.msvc19/conan_toolchain.cmake:31 (message):
  The CMake policy CMP0091 must be NEW, but is ''
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.23/Modules/CMakeDetermineSystem.cmake:121 (include)
  CMakeLists.txt:2 (project)


-- Configuring incomplete, errors occurred!
```

**Comment by ximinez** [comment]
> what version of CMake are you using?

@thejohnfreeman 
```
cmake version 3.24.2
```

**Review by ximinez — CHANGES_REQUESTED** [review]
See my comment on the corresponding changes in `rippled`: https://github.com/XRPLF/rippled/pull/4831#pullrequestreview-1818643351

File: conanfile.txt
Comment (ximinez) [inline conanfile.txt:2]: This will need to be updated once https://github.com/XRPLF/rippled/pull/4831 is merged.

**Review by ximinez — APPROVED** [review]
With the latest changes from here and https://github.com/XRPLF/rippled/pull/4831, I am able to get a successful build. The dependency will need to be updated once 4831 is merged.

File: conanfile.txt
Comment (intelliot) [reply conanfile.txt:2]: It should be ok to do that in a separate/new PR, right? 

Feel free to create an issue here so that it doesn't get forgotten.
- https://github.com/ripple/validator-keys-tool/issues/new

Comment (ximinez) [reply conanfile.txt:2]: Since XRPLF/rippled#4831 needs to be merged first anyway, I think it would make more sense to do that merge, update the dependency on this PR, and then merge this PR. It wouldn't need another review (or no more than a quick glance), since it's just a version number.

**Comment by ximinez** [comment]
@thejohnfreeman Can you update the `conanfile.txt` to point to xrpl 2.1.0 or cherry pick https://github.com/ximinez/validator-keys-tool/commit/044cce331fe28ce711a0a4c80b5e3355d21579bb from my https://github.com/ximinez/validator-keys-tool/commits/jf/conan/ branch?

**Review by ximinez — CHANGES_REQUESTED** [review]
@thejohnfreeman Can you update the `conanfile.txt` to point to xrpl 2.1.0 or cherry pick https://github.com/ximinez/validator-keys-tool/commit/044cce331fe28ce711a0a4c80b5e3355d21579bb from my https://github.com/ximinez/validator-keys-tool/commits/jf/conan/ branch?

**Review by ximinez — APPROVED** [review]

File: CMakeLists.txt
Comment (legleux) [inline CMakeLists.txt:12]: Do we still want this hanging around? I hope the `rippled` build isn't relying on this?

File: cmake/KeysNIH.cmake
Comment (legleux) [inline cmake/KeysNIH.cmake:1]: Do we still want this?

File: CMakeLists.txt
Comment (thejohnfreeman) [reply CMakeLists.txt:12]: The `rippled` build isn't relying on it. The build of this project, `validator-keys-tool`, relies on it. It is how `include(KeysNIH)` on the next line can find the `KeysNIH.cmake` module. Same for the other `include()`s.

File: cmake/KeysNIH.cmake
Comment (thejohnfreeman) [reply cmake/KeysNIH.cmake:1]: Removed. :+1:

**Comment by legleux** [comment]
:+1:  Built fine for me on macOS and linux. 
VS2022 on windows I still had issues with but I doubt it's related to this project.

**Review by legleux — APPROVED** [review]

File: CMakeLists.txt
Comment (legleux) [reply CMakeLists.txt]: Whoops, what I meant was, "the `validator-keys-tool` _target_ in the rippled repo.
Like, if it's removed, we'll need to change `rippled`.

Comment (thejohnfreeman) [reply CMakeLists.txt]: It's two different directories. Note the `CMAKE_CURRENT_SOURCE_DIR`, which points into this project, not rippled. rippled adds its own module directory in its [listfile](https://github.com/XRPLF/rippled/blob/develop/CMakeLists.txt#L12).

**Pull request merged**
